tests/util/setup-dev-tests.ts

```diff
@@ -199,33 +199,40 @@ export function describeDevMoonbeam(
 
         const { parentHash, finalize } = options;
 
+        
+        // TODO: Removes this whole check once Frontier support block import wait for
+        // create block. (cc @tgmichel)
+
         // We are now listening to the eth block too. The main reason is because the Ethereum
         // ingestion in Frontier is asynchronous, and can sometime be slightly delayed. This
         // generates some race condition if we don't wait for it.
-
-        let expectedBlockNumber: number = (await subProvider.eth.getBlockNumber()) + 1;
-        const ethCheckPromise = new Promise<void>((resolve) => {
-          const ethBlockSub = subProvider.eth
-            .subscribe("newBlockHeaders", function (error, result) {
-              if (!error) {
-                return;
-              }
-              console.error(error);
-            })
-            .on("data", function (blockHeader) {
-              // unsubscribes the subscription once we get the right block
-              if (blockHeader.number != expectedBlockNumber) {
-                debug(
-                  `Received unexpected block: ${blockHeader.number} ` +
-                    `(expected: ${expectedBlockNumber})`
-                );
-                return;
-              }
-              ethBlockSub.unsubscribe();
-              resolve();
-            })
-            .on("error", console.error);
-        });
+        // We don't use the blockNumber because some tests are doing "re-org" which would make
+        // the new block number not to be the expected one.
+        let currentBlockHash = (await subProvider.eth.getBlock("latest")).hash;
+        const ethCheckPromise = parentHash
+          ? Promise.resolve()
+          : new Promise<void>((resolve) => {
+              const ethBlockSub = subProvider.eth
+                .subscribe("newBlockHeaders", function (error, result) {
```

Comment:
Is the goal of this to fix the race condition on ethereum data availability? `newBlockHeaders` internally triggers when there is a substrate block import notification, so there is still no guarantee that is mapped in the secondary database.

Reply:
Yes it was the goal and it seems to work, so either it is a side effect or not... but we are planning to remove it anyway